### ISSUE TITLE
Add configurable option to disable all phone home requests

### ISSUE THREAD

**Issue by jsmilani** [webmin/webmin#1213]
I think it is a good practice to allow disabling phone home behavior in applications. I switched off every phone home setting I could find and scoured code for what conditions and many are just not configurable.

The response "there is nothing malicious" isn't the point. On high security private networks where data must be protected it is common to block outside network connections and even raise alarms when this happens. Truth is there have been malicious code injected in webmin in the past (https://www.ceos3c.com/news/webmin-backdoor-found-one-year-after-hackers-released-it/) so just because there is nothing malicious now doesn't mean there couldn't be in the future. That doesn't mean malicious code would respect these options but understanding that some organizations must guard against it should be taken into account. In my situation in particular our firewall doesn't close connections. Yes, this is stupid, unexpected behavior, but it is our reality. The result is the Dashboard doesn't load until the connections timeout (the timeout is not configurable either) and the app just seems broken. It does eventually load if I leave the browser window open but it is also keeping the forked processes running in the background the whole time and could lead to memory issues or running out if PIDs.

This is somewhat related to https://github.com/webmin/webmin/issues/993 and that helped me identify the hostnames that it was connecting to. I don't have direct control over the host firewall since we run inside docker but the simple workaround in my case was to add the following to my docker-compose file:

```
services:
  my_service:
    ...
    extra_hosts:
      - "api.github.com:127.0.0.2"
      - "announce.webmin.com:127.0.0.2"
      - "www.webmin.com:127.0.0.2"
      - "download.webmin.com:127.0.0.2"
```

Figuring this out took about 10 hours of trying different things to get it to work so a flag to disable all of these phone home requests would have been a big time saver.

**Comment by chris001** [comment]
@jsmilani  You do realize that, if you were to disable all "phone home" requests, you'd also fail to see the critical security updates announcements on the main Webmin dashboard, and subsequently, your server could be compromised/ransomwared, and data stolen/breached by malicious parties, without you becoming aware by means of the announcement that you really must update immediately to prevent data breach/ransom situation.  Up to you!

**Comment by jcameron** [comment]
@jsmilani - would this include checks for updated packages using APT or YUM? We could turn off the other phone-homes, but as @chris001 says turning off checks for package updates is likely to make you less secure.

**Comment by PitWenkin** [comment]
Should the package update check via APT/YUM not be executed outside of webmin itself, together with all other system package update checks.
Except for ease of use/updating via GUI, there is no real need for webmin to check them itself.

**Comment by jcameron** [comment]
Right, all Webmin does for packages is call YUM or APT to get available updates.

**Comment by jsmilani** [comment]
We run webmin in docker and we treat docker containers as immutable. I am sure we are diverging a great deal from what is considered a "standard installation" for webmin but this is best practice for using Docker with an Immutable Architecture (https://www.hashicorp.com/resources/what-is-mutable-vs-immutable-infrastructure). In an Immutable Architecture you don't update things in place and docker makes that easy by swapping out the entire service with a new docker image. This does introduce issues such as how to keep things up-to-date as you mentioned, but you don't always want the latest unless you know it is stable and compatible so the process around building new images must be more structured and vulnerability detection software running periodically will be a forcing function for upgrades. Anyway we are not going to rely on the tools themselves telling us when they need updates so we really have no use for a phone home feature like this.

**Comment by swelljoe** [comment]
Reviving this old issue before closing, if the dashboard still requires external access or becomes unusably slow, that's a bug that should be fixed.

@iliaross is the dashboard still waiting on these requests? That's a performance disaster, even if they aren't blocked. It's unreasonable to make a bunch of web requests on the backend during dashboard loading.

**Comment by iliaross** [comment]
> [@iliaross](https://github.com/iliaross) is the dashboard still waiting on these requests?

Everything should be cached, and the dashboard should load instantly. All requests are processed in the background. The only time performance may slow down is during manual cache purging and visiting the dashboard right away.

I should probably take care of that too.
